solidity/contracts/examples/Consumer.sol

```diff
@@ -0,0 +1,41 @@
+pragma solidity ^0.4.21;
+
+import "../Chainlinked.sol";
+
+contract Consumer is Chainlinked {
+  bytes32 internal requestId;
+  bytes32 public currentPrice;
+
+  function Consumer(address _link, address _oracle)
+    public
+  {
+    setLinkToken(_link);
+    setOracle(_oracle);
+  }
+
+  function requestEthereumPrice(string _currency)
+    public
+  {
+    ChainlinkLib.Run memory run = newRun("someJobId", this, "fulfill(bytes32,bytes32)");
+    run.add("url", "https://etherprice.com/api");
+    string[] memory path = new string[](2);
+    path[0] = "recent";
+    path[1] = _currency;
+    run.addStringArray("path", path);
+    requestId = chainlinkRequest(run);
```

Comment:
I know we originally talked about having all methods on the `ChainlinkLib.Run`, but would it make sense to separate `Buffer` from the CL specific material? i.e:

```
    Buffer memory buf = new Buffer(128);
    buf.add("url", "https://etherprice.com/api");
    requestId = chainlinkRequest(
      buf,
      "someJobId",
      this,
      "fulfill(bytes32,bytes32)"
    );

```

Reply:
Let's move run.id generation into `chainlinkRequest`